src/Bicep.Local.Deploy/Extensibility/LocalExtensionHost.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+using System.Threading.Tasks;
+using Azure.Deployments.Extensibility.Core.V2.Json;
+using Bicep.Local.Extension.Protocol;
+
+namespace Bicep.Local.Deploy.Extensibility
+{
+    public class LocalExtensionHost : ILocalExtensionHost
+    {
+        private readonly ILocalExtensionFactoryManager extensionFactoryManager;
+
+        public LocalExtensionHost(ILocalExtensionFactoryManager extensionFactoryManager)
+        {
+            this.extensionFactoryManager = extensionFactoryManager;
+        }
+
+        public async Task<HttpResponseMessage> CallExtensibilityHost(LocalDeploymentEngineHost.ExtensionInfo extensionInfo, HttpContent content, CancellationToken cancellationToken)
+        {
+            var extension = await extensionFactoryManager.GetLocalExtensionAsync(extensionInfo.ExtensionName, extensionInfo.ExtensionVersion);
```

Comment:
Calling `extensionFactoryManager` ensures that we load the extension on-demand and only once.